### PR TITLE
Windows Node Group Pattern

### PR DESCRIPTION
**Pull request by elamaran11**
*Issue #, if available:*

*Description of changes:*
Windows Node Group Pattern

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by elamaran11** [comment]
@shapirov103 Could we merge this.

File: docs/patterns/windows.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
+# Windows Nodes on EKS
+
+Legacy applications in many industries tend to run on Windows. Customers want to scale these workloads on Kubernetes alongside their Linux workloads. Amazon EKS supports windows node groups and you can Windows worker node group to an Amazon EKS cluster. This pattern will deploy an Creates EKS Cluster Control plane with a managed node group running windows node. Please check our AWS doc on [Enabling Windows support for your Amazon EKS cluster](https://docs.aws.amazon.com/eks/latest/userguide/windows-support.html) to learn more about considerations, prerequisites in running windows nodes with EKS cluster. Also please reference this AWS doc to learn about [Amazon EKS optimized Windows AMIs](https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-windows-ami.html).
+
+### Addons
+Not all of the listed EKS addons support windows. We are currently working a list of supported addons documentation which will be available at [documentation](https://github.com/aws-quickstart/cdk-eks-blueprints/blob/main/docs/addons/index.md).
```
Comment (shapirov103) [inline docs/patterns/windows.md:6]: We are currently working a list => We are currently working on a list

Comment (shapirov103) [inline docs/patterns/windows.md:6]: available at [documentation] => published [here]

**Review by shapirov103 — COMMENTED** [review]
Two awesome patterns here, great work! A few comments below.

**Comment by shapirov103** [comment]
> @shapirov103 Could we merge this.

@elamaran11 will gladly merge, let's address the feedback but the PR is quite ripe for merging.

File: docs/patterns/windows.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
+# Windows Nodes on EKS
+
+Legacy applications in many industries tend to run on Windows. Customers want to scale these workloads on Kubernetes alongside their Linux workloads. Amazon EKS supports windows node groups and you can Windows worker node group to an Amazon EKS cluster. This pattern will deploy an Creates EKS Cluster Control plane with a managed node group running windows node. Please check our AWS doc on [Enabling Windows support for your Amazon EKS cluster](https://docs.aws.amazon.com/eks/latest/userguide/windows-support.html) to learn more about considerations, prerequisites in running windows nodes with EKS cluster. Also please reference this AWS doc to learn about [Amazon EKS optimized Windows AMIs](https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-windows-ami.html).
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply docs/patterns/windows.md]: Fixed,

[the remark this answers is not in the document]
Comment (elamaran11) [reply docs/patterns/windows.md]: Fixed

File: docs/patterns/windows.md (hunk 1 above)
Comment (elamaran11) [reply docs/patterns/windows.md]: Fixed

Comment (elamaran11) [reply docs/patterns/windows.md]: fixed

File: docs/patterns/windows.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (elamaran11) [reply docs/patterns/windows.md]: fixed

[the remark this answers is not in the document]
Comment (elamaran11) [reply docs/patterns/windows.md]: fixed

File: lib/common/windows-builder.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+                    role: blueprints.getResource(context => {
+                        return new iam.Role(context.scope, 'ClusterRole', { 
+                            assumedBy: new iam.ServicePrincipal("eks.amazonaws.com"),
+                            managedPolicies: [
+                                iam.ManagedPolicy.fromAwsManagedPolicyName("AmazonEKSClusterPolicy"),
+                                iam.ManagedPolicy.fromAwsManagedPolicyName("AmazonEKSVPCResourceController")
+                            ]
+                        });
+                    }),
+                    mastersRole: blueprints.getResource(context => {
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply lib/common/windows-builder.ts:50]: I think we discussed about this couple of weeks ago. In Windows Node, Control Plane needs the VPC Controller Policy without with Windows nodes wont work and hence we need this. Once we have framework support this, we can remove or tinker this.

File: lib/windows-construct/index.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            clusterProviderTags: {
+                "Name": "blueprints-windows-eks-cluster",
+                "Type": "generic-windows-cluster"
+            },
+            genericNodeGroupTags: {
+                "Name": "Mng-linux",
+                "Type": "Managed-linux-Node-Group",
+                "LaunchTemplate": "Linux-Launch-Template",
+            },
+            windowsNodeGroupTags: {
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply lib/windows-construct/index.ts:41]: Agreed. We can defer this to Framework PR and we have an issue for it already.

**Comment by elamaran11** [comment]
> 

@shapirov103 All issues are addressed and fixed, please review and Merge.

File: lib/common/windows-builder.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (shapirov103) [reply lib/common/windows-builder.ts:50]: We discussed the cluster role. My comment is specific to the line 50, mastersRole. What are you adding here that is not provided ootb by the blueprints?

**Review by shapirov103 — COMMENTED** [review]
Please see my comment.

**Comment by elamaran11** [comment]
> Please see my comment.

@shapirov103 Yes you are right. ClusterRole is what i was talking. This one is not required, didnt remove it.

**Comment by elamaran11** [comment]
All Lint issues fixed, Please check .

**Pull request closed without merging**